tensorzero-core/src/variant/dicl.rs

```diff
@@ -620,7 +686,49 @@ impl DiclConfig {
         })
     }
 
-    fn prepare_request<'a, 'request>(
+    /// Prepare request message for DICL (stringify template content blocks, then delegate to chat_completion)
+    async fn prepare_request_message_dicl(
+        message: &LazyResolvedInputMessage,
+        templates: &TemplateConfig<'_>,
+    ) -> Result<RequestMessage, Error> {
+        let transformed_content: Vec<LazyResolvedInputMessageContent> = message
+            .content
+            .iter()
+            .map(
+                |content_block| -> Result<LazyResolvedInputMessageContent, Error> {
+                    match content_block {
+                        LazyResolvedInputMessageContent::Template(template_input) => {
+                            // Stringify the template as JSON since DICL variants don't have a concept of templates
+                            let json_str = serde_json::to_string(&serde_json::json!({
+                                "type": "template",
+                                "name": template_input.name,
+                                "arguments": template_input.arguments,
+                            }))
+                            .map_err(|e| {
+                                Error::new(ErrorDetails::Serialization {
+                                    message: format!(
+                                        "Failed to stringify template content block: {e}"
+                                    ),
+                                })
+                            })?;
+                            Ok(LazyResolvedInputMessageContent::Text { text: json_str })
+                        }
```

Comment:
I don't really understand why we need the name as part of the stringification here

Reply:
If I have two templates with same/similar variables the model should know what I'm referring to